erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -419,6 +419,12 @@ def set_missing_values(self, for_validate=False):
 
 		super(SalesInvoice, self).set_missing_values(for_validate)
 
+		if not self.get('payment_terms_template'):
```

Comment:
I think we should move this to `set_missing_values` of `make_sales_invoice` under `delivery_note.py`.

As it is being followed in purchase cycle

Reply:
Made the changes